lib/modules/apostrophe-schemas/src/apos/components/ApostropheStringField.vue

```diff
@@ -8,28 +8,13 @@
 
 <script>
 
+import ApostropheFieldMixin from './ApostropheFieldMixin.js';
+
 export default {
+  mixins: [ ApostropheFieldMixin ],
   name: 'ApostropheStringField',
-  props: {
-    value: Object,
-    field: Object
-  },
-  data() {
-    return {
-      next: (this.value.data !== undefined) ? this.value.data : (this.field.def || ''),
-      error: false
-    };
-  },
-  mounted() {
-    this.updateErrorAndEmit();
-  },
-  watch: {
-    next(value) {
-      this.updateErrorAndEmit();
-    }
-  },
   methods: {
-    updateErrorAndEmit() {
+    validate() {
```

Comment:
this should return error (or false) rather than setting this.error directly.

Reply:
Returns `this.error`. See below.